boot/boot.js

```diff
@@ -287,6 +287,136 @@ $tw.utils.decodeURIComponentSafe = function(s) {
 	return v;
 };
 
+/*************
+Code to make the permalink/permaview URLs more readable.
+(These comments are just to help reviewers; they will eventually be removed.)
+***************/
+
+
+/*************
+Helpers for encoding/decoding URIs.  **Note** that these are throw-away, only
+used to temporarily build `charMatch`, `spaceMatch`, `sentenceEnding`, and
+`sentenceTrailing`, all of which will eventually be hard-coded.
+***************/
+
+/*
+The character that will substitute for a space in the URL
+*/
+var SPACE = "_";
+
+/*
+The character added to the end to avoid ending with `.`, `?`, `!` or the like
+*/
+var TRAILER = '_';
+
+/*
+The character that will separate out the list elements in the URL
+*/
+var CONJUNCTION = ";";
+
+/*
+Those of the allowed url characters claimed by TW
+*/
+var CLAIMED = [SPACE, ":", CONJUNCTION];
+
+/*
+Non-alphanumeric characters allowed in a URL fragment
+More information at https://www.rfc-editor.org/rfc/rfc3986#appendix-A
+*/
+var VALID_IN_URL_FRAGMENT = "-._~!$&'()*+,;=:@/?";
+
+/*
+The subset of the pchars we will not percent-encode in permalinks/permaviews
+*/
+var substitutes = VALID_IN_URL_FRAGMENT.split('').filter(function(c){return CLAIMED.indexOf(c) === -1});
+
+
+/*************
+The more permanent implementation.  These helpers (`charMatch`, `spaceMatch`,
+`sentenceEnding`, `sentenceTrailing`) will be hard-coded, not derived.
+***************/
```

Comment:
IMO this comment **above** can go away
The comment **below** should stay **as is**